include/experimental/__p1673_bits/blas2_matrix_vector_product.hpp

```diff
@@ -350,9 +525,9 @@ void triangular_matrix_vector_product(in_matrix_t A,
     }
   }
   else {
-    for (ptrdiff_t j = 0; j < A.extent(1); ++j) {
-      const ptrdiff_t i_upper = explicitDiagonal ? j : j - ptrdiff_t(1);
-      for (ptrdiff_t i = 0; i <= i_upper; ++i) {
+    for (size_type j = 0; j < A.extent(1); ++j) {
+      const ptrdiff_t i_upper = explicitDiagonal ? j : j - size_type(1);
```

Comment:
The difference between two `size_t` is (unfortunately) still `size_t`, so this might wrap around.

https://godbolt.org/z/aMjMcKMEW